robot/game.py

```diff
@@ -36,12 +36,10 @@ def zone(self):
         return self.send_and_receive({})['zone']
 
     @property
-    def mode(self):
+    def mode(self) -> GameMode:
         """
         Whether or not the robot is in competition mode
         :return: if the robot is in competition mode
         """
         value = self.send_and_receive({})['mode']
-        for enum in GameMode:
-            if value == enum.value:
-                return enum
+        return GameMode(value)
```

Comment:
Please can we _not_ edit functionality (even when it's equivalent) as part of this PR? (The easiest way to do this is to cherry-pick the fixes onto master separately and then merge master outwards into this branch) This makes it clearer what changes are being made.

Alternatively, please can we make it clear in the PR (and merge message) that this will also fix issues essential to making the types work.